### PR TITLE
feat(bigquery): add support of use_avro_logical_types for extract jobs

### LINKED ISSUE #9624: BigQuery: Add support for use_avro_logical_types for extract jobs
use_avro_logical_types is supported for load jobs: https://github.com/googleapis/google-cloud-python/blob/89eaedb4a40bd1268df134cc40a4063d6cf8b823/bigquery/google/cloud/bigquery/job.py#L1318

We should add the same mapping for extract jobs. BigQuery already supports it in the API backend.

### PR DESCRIPTION
**Pull request by HemangChothani**
Fixes #9624

### CODE REVIEW

**Comment by HemangChothani** [comment]
kokoro - BigQuery failure related to [9633](https://github.com/googleapis/google-cloud-python/issues/9633) issue.

File: bigquery/google/cloud/bigquery/job.py [hunk 1]
```diff
@@ -1910,6 +1910,18 @@ def print_header(self):
     def print_header(self, value):
         self._set_sub_prop("printHeader", value)
 
+    @property
+    def use_avro_logical_types(self):
+        """bool: For loads of Avro data, governs whether Avro logical types are
+        converted to their corresponding BigQuery types(e.g. TIMESTAMP) rather than
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/job.py]: Nit: Missing a space between `types` and `(e.g.`

Comment (plamut) [reply bigquery/google/cloud/bigquery/job.py]: Let's then fix the original docstring, too (in `LoadJobConfig.use_avro_logical_types`).

**Comment by plamut** [comment]
@HemangChothani Thanks for the cleanup, let's add it to the original `use_avro_logical_types()` method, too (boy scouts and such).

**Pull request merged**